islands/IssuesList.tsx

```diff
@@ -0,0 +1,80 @@
+import { GitHubIssue } from '../lib/github.ts';
+import Card from '../components/Card.tsx';
+import { useState } from 'preact/hooks';
+
+type Props = {
+  repos: Array<string>;
+  issues: Array<GitHubIssue>;
+};
+
+type LabelProps = {
+  name: string;
+  color: string;
+}
+
+const IssuesList = ({repos, issues}: Props) => {
+  const [repoFilter, setRepoFilter] = useState(null);
+  const [labelFilter, setLabelFilter] = useState({});
```

Comment:
You can tell TypeScript what the types of these state values will be, which is good so in the places the values are used, TypeScript will have a narrower set of possible types to account for and can avoid some compatibility issues.

Also any place you're wanting to represent an "empty" type value, opting for `null` or `undefined` is usually best. In the case of `repoFilter`, the reason a `NO_FILTER` const is needed here is because the `option` element doesn't support `null` as a value, and `undefined` means the `value` defaults to the string text inside the `option` (`"Filter by Repo" in this case) 😅

TypeScript is crazy but there are usually good reasons why the types are defined the ways they are! In this case it's a representation of the types that are expected by the elements in the browser, and going outside those expected types can lead to unexpected behavior.

```suggestion
  const NO_FILTER = "none";
  const [repoFilter, setRepoFilter] = useState<string>(NO_FILTER);
  const [labelFilter, setLabelFilter] = useState<LabelProps | undefined>();
```